app/views/posts/show.html.erb

```diff
@@ -8,6 +8,10 @@
 
   </div>
   <div style="display: flex; padding: 6px; gap: 10px">
+    <% if can? :destroy, @post %>
+    <%= button_to "Delet This Post" ,[@post.author, @post], method: :delete %>
+    <% end %>
+
     <p>Comments: <%= @post.comments_counter %></p>
     <p>Likes: <%= @post.likes_counter %></p>
   </div>
```

Comment:
- [ ] I created a user and made them `admin`, now when I try to visit this post show page, I get an access denied error. Kindly note that every user, regardless of their role, should be able to `read` the pages, but only the admin and post author should be able to see the delete button for comment and posts and delete them. Kindly fix this.

  ![image](https://github.com/bhushan354/Blog-App-Rails/assets/36542986/1673956d-707b-4da1-b02f-978d5b626c9a)